src/transformers/tokenization_mistral_common.py

```diff
@@ -0,0 +1,354 @@
+import os
+import shutil
+from pathlib import Path
+from typing import TYPE_CHECKING, Callable, Dict, List, Optional, Tuple, Union
+
+from mistral_common.protocol.instruct.request import ChatCompletionRequest
+from mistral_common.protocol.instruct.validator import ValidationMode
+from mistral_common.tokens.tokenizers.base import SpecialTokenPolicy
+from mistral_common.tokens.tokenizers.mistral import MistralTokenizer
+from mistral_common.tokens.tokenizers.utils import download_tokenizer_from_hf_hub
+
+from transformers.tokenization_utils_base import (
+    PreTrainedTokenizerBase,
+    TextInput,
+)
+from transformers.utils import logging, to_py_obj
+from transformers.utils.hub import PushToHubMixin
+from transformers.utils.import_utils import is_tf_available, is_torch_available
+
+
+if TYPE_CHECKING:
+    if is_torch_available():
+        import torch
+    if is_tf_available():
+        import tensorflow as tf
+    import numpy as np
+
+
+logger = logging.get_logger(__name__)
+
+
+class MistralCommonTokenizer(PushToHubMixin):
+    """
+    Class to wrap mistral-common tokenizers.
+
+    This class is a wrapper around a `mistral_common.tokens.tokenizers.mistral.MistralTokenizer`.
+    It provides a convenient interface to tokenize using the official mistral-common tokenizer.
+
+    Handles several methods for tokenization as well as methods for
+    downloading/caching/loading pretrained tokenizers.
+    """
+    def __init__(
+        self, tokenizer_path: Union[str, os.PathLike, Path], mode: ValidationMode = ValidationMode.test, **kwargs
+    ):
+        """
+        Initiales a mistral-common tokenizer.
+
+        Args:
+            tokenizer_path (`Union[str, os.PathLike, Path]`):
+                Path to the tokenizer file.
+            mode (`ValidationMode`, *optional*, defaults to `ValidationMode.test`):
+                Validation mode for the tokenizer.
+            **kwargs (additional keyword arguments, *optional*):
+                Not supported by `MistralCommonTokenizer`.
+                Will raise an error if used.
+        """
+        if kwargs:
+            raise ValueError(f"Kwargs {list(kwargs.keys())} are not supported to init `MistralCommonTokenizer`.")
```

Comment:
```suggestion
            raise ValueError(f"Kwargs {list(kwargs.keys())} are not supported to init `MistralCommonTokenizer`.")
```

maybe a warning instead?

Reply:
Agreed that it could be a warning, let's see what HF folks suggest in the transformers PR